### PR TITLE
More locations and changes to solar system detection

### PR DESCRIPTION
**Pull request by scimas**

### CODE REVIEW

File: library/lib_location_constants.ks [hunk 1]
```diff
@@ -5,7 +5,7 @@ global location_constants is lex().
 //runways should be added in this pattern "name_runway_##_start" where name is the name of the runway and ## is the runway number
 //  If not added with this pattern then the automatic alias creation that add an end position based on the start of the same name but opposite number will fail
 
-if bodyexists("kerbin") and bodyexists("mun") and bodyexists("minmus") {//check for if the craft is in the stock solar system
+if bodyexists("Sun") and body("Sun"):radius = 261_600_000 {//check for if the craft is in the stock solar system
```
Comment (charlesstaats) [inline library/lib_location_constants.ks]: I think it's probably wise to include `bodyexists("kerbin")` as well before using code that references `Kerbin`.

Side note: I somehow missed or forgot that you could add in underscore separators in scalar literals. Definitely improves readability!

File: library/lib_location_constants.ks [hunk 2]
```diff
@@ -34,6 +34,41 @@ if bodyexists("kerbin") and bodyexists("mun") and bodyexists("minmus") {//check
...
+
+  location_constants:add("runway_start",kerbinLocations["runway_09_start"]).
+  location_constants:add("reverse_runway_start",kerbinLocations["runway_27_start"]).
+
+  location_constants:add("kerbin",kerbinLocations).
+}
+
+// RealSolarSystem locations
+if bodyexists("Sun") and body("Sun"):radius = 696_342_000 {
+  local earthLocations is lex().
```
Comment (charlesstaats) [inline library/lib_location_constants.ks]:
```suggestion
  local earthLocations is lex().
  // TODO: populate at least the most commonly used of the RealSolarSystem locations.
```

**Review by charlesstaats — APPROVED** [review]
A couple suggestions. Thanks for adding all the JNSQ locations!

File: library/lib_location_constants.ks (hunk 1 above)
Comment (scimas) [reply library/lib_location_constants.ks]: Good point, done

**Comment by charlesstaats** [comment]
@scimas I'm in the process of pulling and "testing" the PR (at least enough to be sure running it won't crash stock). Should I wait for any more commits before doing so?

**Comment by scimas** [comment]
No, this should be enough for now, barring any discovered bugs.

**Pull request merged**